payment_source.go

```diff
@@ -0,0 +1,67 @@
+package stripe
+
+import (
+	"encoding/json"
+	"github.com/stripe/stripe-go/paymentsource"
+)
+
+// SourceParams is the set of parameters that can be used to describe the source
+// object used to make a Charge. For example, both Card and BitcoinReceiver objects
+// can be described by SourceParams.
+// For more details see https://stripe.com/docs/api#create_charge
+type SourceParams struct {
+	Params
+	ID    string
+	Token string
+	Card  *CardParams
+}
+
+// Interface to provide a human readable representation of a struct
+type Displayer interface {
+	Display() string
+}
+
+// PaymentSource describes the payment source used to make a Charge.
+// The Type should indicate which object is fleshed out (eg. BitcoinReceiver or Card)
+// For more details see https://stripe.com/docs/api#retrieve_charge
+type PaymentSource struct {
```

Comment:
Yeah -- I didn't really love that in order to access the common properties ("id" & "object"), end users would need a type assertion or we would provide a bunch of redundant getters to satisfy the interface. I also got to move the unmarshaling of the Card/BitcoinReceiver structs into the PaymentSource itself out of Charge, which ~broke some level of encapsulation (https://github.com/stripe/stripe-go/commit/20d58c60d5bac016bb3a13d5775443d3f1c2ad06#diff-04f5f707872b10c0ab2a591bceddbf5eL79). I may well have missed a better approach to unmarshaling into an interface property, though.